test/cppsuite/test_harness/core/component.h

```diff
@@ -34,8 +34,10 @@
 
 namespace test_harness {
 /*
- * A component is a class that defines 3 unique stages in its life-cycle, the stages must be run in
- * the following order: load, run, finish.
+ * A component is a class that defines 4 unique stages in its life-cycle, the stages must be run in
+ * the following order: load, run, end_run, finish.
+ *
+ * end_run informs the component that is no longer running which closes out its run loop.
```

Comment:
Can we have this comment line 65 instead? We could add a comment on top of `do_work` as well, the same way we did for `load` and `run`.